### DISCUSSION TITLE
Is it possible to link users with different providers?

### DISCUSSION THREAD

**Discussion by asyncink** [supabase/supabase#5283] [Questions]
Hello!

Is it possible to use different providers to login with the same supabase user? For example, user registered with SMS using MessageBird provider. We want to give him ability to link email, Apple and Google accounts (both) in our profile section. So this user can link these providers in profile section and then login with any of the them.

**Comment by GaryAustin1** [comment]
Looks like it just works... at least for email based accounts.
https://github.com/supabase/supabase/discussions/3098

**Reply by asyncink** [reply]
Thank you for your reply!

I think it is rather different case.

Let me explain further:

1. User signs up with his phone number. 
2. User purchases an item, so we have a purchase record in his profile now.
3. Then several months later the same user comes again to our website, but now he signs up with his Google account (he forget about his phone account, for example).
4. User purchases another item and for now he has two different accounts with one purchase record in each account. 

So that is the problem. The workaround here is to give possibility to link another providers to his phone account during step 1. So he signs up with phone and we ask him to link Google provider as well, so he can sign in with both providers (phone and Google) to the same account. Hi he comes later and signs in with Google, he still can see his previous purchase and account.

**Reply by GaryAustin1** [reply]
Hopefully others come along as I can't help with more than basic auth stuff.
